src/pages/home/sidebar/SignInOrAvatarWithOptionalStatus.js

```diff
@@ -0,0 +1,110 @@
+/* eslint-disable rulesdir/onyx-props-must-have-default */
+import React, {useCallback} from 'react';
+import {View} from 'react-native';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import lodashGet from 'lodash/get';
+import withCurrentUserPersonalDetails from '../../../components/withCurrentUserPersonalDetails';
+import PressableWithoutFeedback from '../../../components/Pressable/PressableWithoutFeedback';
+import personalDetailsPropType from '../../personalDetailsPropType';
+import PressableAvatarWithIndicator from './PressableAvatarWithIndicator';
+import Button from '../../../components/Button';
+import Text from '../../../components/Text';
+import styles from '../../../styles/styles';
+import Navigation from '../../../libs/Navigation/Navigation';
+import * as Session from '../../../libs/actions/Session';
+import Permissions from '../../../libs/Permissions';
+import compose from '../../../libs/compose';
+import useLocalize from '../../../hooks/useLocalize';
+import ONYXKEYS from '../../../ONYXKEYS';
+import ROUTES from '../../../ROUTES';
+import CONST from '../../../CONST';
+
+const propTypes = {
+    /** The personal details of the person who is logged in */
+    currentUserPersonalDetails: personalDetailsPropType,
+
+    /** Whether the create menu is open or not */
+    isCreateMenuOpen: PropTypes.bool,
+
+    /** Beta features list */
+    betas: PropTypes.arrayOf(PropTypes.string),
+};
+
+const defaultProps = {
+    betas: [],
+    isCreateMenuOpen: false,
+    currentUserPersonalDetails: {
+        status: {emojiCode: ''},
+    },
+};
+
+function SignInOrAvatarWithOptionalStatus({currentUserPersonalDetails, isCreateMenuOpen, betas}) {
+    const {translate} = useLocalize();
+    const statusEmojiCode = lodashGet(currentUserPersonalDetails, 'status.emojiCode', '');
+    const emojiStatus = Permissions.canUseCustomStatus(betas) ? statusEmojiCode : '';
+
+    const showStatusPage = useCallback(() => {
+        if (isCreateMenuOpen) {
+            // Prevent opening Settings page when click profile avatar quickly after clicking FAB icon
+            return;
+        }
+
+        Navigation.setShouldPopAllStateOnUP();
+        Navigation.navigate(ROUTES.SETTINGS_STATUS);
+    }, [isCreateMenuOpen]);
+
+    if (Session.isAnonymousUser()) {
```

Comment:
I also think that the JSX returned in these two if's (line 57 and 75) could be extracted to separate components, to separate concerns - what do you think?

Reply:
I haven't refactored all the components, but moving it outside makes sense. I'll update it. Thanks for pointing it up!